### ISSUE TITLE
arm32v6 image failing to build on RPi ZeroW - Illegal instruction (core dumped)

### ISSUE THREAD

**Issue by rgee0** [docker-library/golang#184]
Trying to perform a multi stage build from `arm32v6/golang:1.9.2-alpine3.6` on an RPi ZeroW results in a core dump at the first `go` call.

OS:
```
cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 8 (jessie)"
NAME="Raspbian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

Docker:
```
Client:
 Version:      17.10.0-ce
 API version:  1.33
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:11:55 2017
 OS/Arch:      linux/arm

Server:
 Version:      17.10.0-ce
 API version:  1.33 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:05:34 2017
 OS/Arch:      linux/arm
 Experimental: false
```

Dockerfile:
```dockerfile
FROM arm32v6/golang:1.9.2-alpine3.6 AS build

ENV GOPATH=/go

RUN go get -d -u github.com/alexellis/blinkt_go/sysfs/gpio

RUN mkdir -p /go/src/github.com/alexellis/blink_go_examples/example
WORKDIR /go/src/github.com/alexellis/blinkt_go_examples/example

COPY . .

RUN GOARM=6 CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /usr/bin/app .

FROM scratch

COPY --from=build /usr/bin/app /app

CMD ["/app"]

```

Output from `docker build .`
```
Sending build context to Docker daemon  8.192kB
Step 1/9 : FROM arm32v6/golang:1.9.2-alpine3.6 AS build
1.9.2-alpine3.6: Pulling from arm32v6/golang
Digest: sha256:2075c33b0adb86a5625efae7676296ab1d9a08bdd1b0a7308a89080583f18901
Status: Downloaded newer image for arm32v6/golang:1.9.2-alpine3.6
 ---> 14f40308ee24
Step 2/9 : RUN go get -d -u github.com/alexellis/blinkt_go/sysfs/gpio
 ---> Running in 7b407f1d7ec4
Illegal instruction (core dumped)
The command '/bin/sh -c go get -d -u github.com/alexellis/blinkt_go/sysfs/gpio' returned a non-zero code: 132
```

**Comment by tianon** [comment]
Doh, this looks hairy!

Can you reproduce the failure with Alpine's "go"package, or is it something
specific to how this image builds Go? (I don't have a Pi Zero or 1, so
can't easily test this.)

**Comment by dsbaha** [comment]
I also wanted to chime in and say that it happens on the Raspberry Pi B+ as well (armv6).  Any go command results in an Illegal instruction (core dumped).  Can't do a go version.

Illegal instruction (core dumped)
The command '/bin/sh -c go get -d -v -t' returned a non-zero code: 132

Utilizing arm32v6/golang:alpine

**Comment by tianon** [comment]
Hmm, this might be part of the problem: (not 100% sure though, especially since I don't have an `arm32v6` device to test with)
```console
$ docker run -it --rm arm32v6/alpine
/ # apk add --no-cache go
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/main/armhf/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.6/community/armhf/APKINDEX.tar.gz
(1/13) Installing binutils-libs (2.28-r3)
(2/13) Installing binutils (2.28-r3)
(3/13) Installing gmp (6.1.2-r0)
(4/13) Installing isl (0.17.1-r0)
(5/13) Installing libgomp (6.3.0-r4)
(6/13) Installing libatomic (6.3.0-r4)
(7/13) Installing pkgconf (1.3.7-r0)
(8/13) Installing libgcc (6.3.0-r4)
(9/13) Installing mpfr3 (3.1.5-r0)
(10/13) Installing mpc1 (1.0.3-r0)
(11/13) Installing libstdc++ (6.3.0-r4)
(12/13) Installing gcc (6.3.0-r4)
(13/13) Installing go (1.8.4-r0)
Executing busybox-1.26.2-r5.trigger
OK: 213 MiB in 24 packages
/ # go version
go version go1.8.4 linux/arm
/ # go env GOARM

/ # 
```
(where that _should_ be explicitly set to `6` instead, at least for our build)

**Comment by tianon** [comment]
There's a good chance https://github.com/docker-library/golang/pull/187 is the appropriate fix for this, but as I said before I don't have a good way to verify it, so if anyone could build that PR on some arm64 hardware and try running it on arm32v6, that'd be helpful (otherwise we'll need to wait for the full release process to get that PR up to `arm32v6/golang` for someone to test it and verify whether it fixes this).

**Comment by alexellis** [comment]
We can verify if you need help.

**Comment by rgee0** [comment]
@tianon I've successfully built [1.9/alpine3.6/Dockerfile](https://github.com/docker-library/golang/pull/187/files#diff-5b7cd08c544ab1c02bdf9beb319ef737) as `rgee0/arm32v6-fix:1.9.2` and then successfully built the earlier failing Dockerfile using `rgee0/arm32v6-fix:1.9.2` as the base. 

I was using a pizerow for verification.

**Comment by yosifkit** [comment]
The fix for this was pushed to Docker Hub with https://github.com/docker-library/official-images/pull/3744.  Feel free to comment if it still core dumps on arm32v6.
